fem/complex_fem.cpp

```diff
@@ -1256,11 +1256,19 @@ ParSesquilinearForm::FormLinearSystem(const Array<int> &ess_tdof_list,
             ess_tdof_list.GetMemory().Read(MemoryClass::DEVICE, n);
          const int *d_diag_i = Aih->diag->i;
          double *d_diag_data = Aih->diag->data;
+#if defined(HYPRE_USING_CUDA)
          CuWrap1D(n, [=] MFEM_DEVICE (int k)
```

Comment:
Since the lambda is the same for both CUDA and HIP, we can avoid code duplication using just:
```c++
 #if defined(HYPRE_USING_CUDA)
          CuWrap1D(n, [=] MFEM_DEVICE (int k)
#elif defined(HYPRE_USING_HIP)
          HipWrap1D(n, [=] MFEM_DEVICE (int k)
#endif
          {
             // ...
          }
```
Or we can define a macro `GpuWrap1D` if this is used in other places too.

Reply:
Done! I defined the macro in capitals though as that is the convection `GPU_WRAP1D`